### PR TITLE
Karatsuba Algorithm In GO

### LINKED ISSUE #2248: Karatsuba Algorithm
- [ ] C - @Angshuman032001
- [x] C++
- [ ] C# - @BishalKrChanda
- [ ] CoffeeScript - @kichloo
- [x] Dart - @heisastark
- [x] Go - @HardevKhandhar
- [ ] Java - @Mustafiz04
- [x] JS - @abhu-A-J
- [ ] Kotlin - @RaghavAwasthi
- [x] Php - @JasbirCodeSpace
- [ ] Python - @lakshyajit165
- [ ] Readme - @lakshyajit165
- [ ] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript - @shivanidalmia1203

### PR DESCRIPTION
**Pull request by HardevKhandhar**
Fixes #2248 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Karatsuba Algorithm In GO
- Dynamic User Input Taken

#### Languages Used:
- GO
- 

#### Files Added:
- Karatsuba_Algorithm.go
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Karatsuba_Algorithm/Karatsuba_Algorithm.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
+// Karatsuba Algorithm In GO
```
Comment (singh-shreya6) [inline Karatsuba_Algorithm/Karatsuba_Algorithm.go:1]: Please add a brief description about the objective of this algorithm

File: Karatsuba_Algorithm/Karatsuba_Algorithm.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    x_high, x_low := getHighAndLowDigits(x, max_digits)
+    y_high, y_low := getHighAndLowDigits(y, max_digits)
+
+    z0 := karatsuba(x_low, y_low)
+    z1 := karatsuba((x_low + x_high), (y_low + y_high))
+    z2 := karatsuba(x_high, y_high)
+
+    if positive {
+        return (z2 * int64(math.Pow(10, float64(2*max_digits)))) + (z1-z2-z0)*int64(math.Pow(10, float64(max_digits))) + z0
```
Comment (singh-shreya6) [inline Karatsuba_Algorithm/Karatsuba_Algorithm.go:89]: Please split this line of code across two lines. It is too big.

File: Karatsuba_Algorithm/Karatsuba_Algorithm.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    } 
+    else {
+        return -((z2 * int64(math.Pow(10, float64(2*max_digits)))) + (z1-z2-z0)*int64(math.Pow(10, float64(max_digits))) + z0)
```
Comment (singh-shreya6) [inline Karatsuba_Algorithm/Karatsuba_Algorithm.go:92]: Please split this line of code across two lines. Also add space around operator in z1-z2-z0 and around *

File: Karatsuba_Algorithm/Karatsuba_Algorithm.go (hunk 2 above)
Comment (singh-shreya6) [reply Karatsuba_Algorithm/Karatsuba_Algorithm.go:89]: Also add space around operators in this (z1-z2-z0)*int64(math.Pow(10, float64(max_digits)))

**Comment by HardevKhandhar** [comment]
Dear Ma'am @singh-shreya6 
The suggested changes have been done. 
The commits have been squashed. 
Thank you.

**Comment by singh-shreya6** [comment]
The commit count is still 2 @HardevKhandhar. Please squash your commits properly so that it is 1.

**Pull request merged**